### DISCUSSION TITLE
[RU] First impression of Taiga

### DISCUSSION THREAD

**Discussion by ghost** [taiga-family/taiga-ui#204] [Ideas]
Добрый день!
Позволю себе написать на русском языке, так как английский не мой родной и не думаю, что смогу точно передать то, о чем хочу сказать.
Так что, надеюсь на понимание :-)

Моё сообщение, это не критика, а впечатление и обратная связь. 

Хочу сказать вам большое спасибо за вашу работу. Впечатляет и видно, что серьезно.

Но, хотелось бы поговорить о том, что не комфортно.
Это конечно же мое субъективное мнение и возможно что то пропустил или пока не понял.

Верстка страницы начинается с макета.
Можно применять абсолютно разные варианты: Angular Flex-Layout, Tailwind CSS, UIkit или вообще написать все самому.
Но когда смотришь на то, что уже есть в Taiga UI, то возникает больше вопросов, чем ответов.
Например, есть Grid, Col, Row, но нет из коробки возможности позиционировать содержимое или "растянуть" по вертикали. Есть marging, но нет padding. И т.д.

Получается, что вроде как поставил какой-нибудь CSS Framework, но часть его уже будет "пересекаться" с тем, что есть в Taiga UI.
Ладно, поставили. Пусть "пересекается".
Но в итоге, когда смотрим теги со стилями сверстанной страницы, происходит "вынос мозга" :-) :
"Вот это" из Taiga UI, а "вот это" из UIkit, а "это" вообще из Angular Flex-Layout...
Но это еще не все.. Есть еще файлы переопределения установленного и еще может быть "что то" свое!

Решать, конечно, вам - менять что то или оставить, как есть.

**Comment by waterplea** [comment]
Привет. Спасибо за отзыв! Дело в том, что мы так закопались в Angular код перед релизом, что не обратили должного внимания на глобальные стили и миксины, которые пришли в библиотеку от верстальщиков на внутренних проектах. На самом деле их надо было просто перенести в проприетарные пакеты и не релизить в опенсорс. Сейчас их уже не убрать, так как это ломающее изменение, но в 3.0 мы обязательно подчистим это дело. Сделаем некоторые глобальные стили опционально подключаемыми, некоторые уберём.

**Comment by vvgrad** [comment]
Не заглядывал пока сильно в код, но думаю ребята делают крутую вещь. Понравились многие UI компоненты. 
Если все пойдет так дальше, думаю Taiga наберёт большую популярность. 
Возможно Вы уже отвечали на эти вопросы, но Вы этим занимаетесь в рамках работы или это отдельный проект, которому Вы уделяете свободное время?  Планируете Вы расширять команду, т.к. судя по видео и комментариям, что видел, задач и пожеланий все больше )

**Reply by waterplea** [reply]
Привет. Спасибо! Этот проект — наша основная работа. По планам на команду — это внутренний кадровый вопрос :)

**Reply by ghost** [reply]
Полностью поддерживаю мнение vvgrad! :-)  Angular реально крут, но по ему субъективному мнению и то, что видел по статистике, он проигрывает в популярности React и Vue исключительно за счет того, что нет классной, хорошо поддерживаемой UI библиотеки компонентов. Пробовал NG-Zorro, PrimeNG, Angular Material - "кривое". Если ребятам удастся создать суперскую, полную библиотеку, закрывающую если не все, то большинство вопросов по созданию интерфейса, а пока вижу, что к этому дело и идет, это будет мировой проект! :-) Удачи! Все получится, болеем за вас! :-)
